### PR TITLE
K23SI Correctness

### PR DESCRIPTION
**Pull request by jfunston**
New inspection RPCs to facilitate tests and correctness tools. Augment tests with new RPCs.

### CODE REVIEW

File: src/k2/dto/K23SIInspect.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+struct K23SIInspectRecordsRequest {
+    // These fields make the request compatible with the PartitionRequest wrapper
+    Partition::PVID pvid; // the partition version ID. Should be coming from an up-to-date partition map
+    String collectionName;
+    Key key; // the key to gather all records for
+    K2_PAYLOAD_FIELDS(pvid, collectionName, key);
+};
+
+struct K23SIInspectRecordsResponse {
+    std::vector<DataRecord> records;
```
Comment (iafuture) [inline src/k2/dto/K23SIInspect.h:45]: I assume this is the DataRecord from 3si module?

File: src/k2/dto/K23SIInspect.h [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    friend std::ostream& operator<<(std::ostream& os, const K23SIInspectTxnResponse& rec) {
+        os << "{txnId=" << rec.txnId << ", writeKeys=[";
+        os << rec.writeKeys.size();
+        os << "], rwExpiry=" << rec.rwExpiry << ", syncfin=" << rec.syncFinalize << "}";
+        return os;
+    }
+};
+
+// For test and debug, requests all WIs on a node for any key
```
Comment (iafuture) [inline src/k2/dto/K23SIInspect.h:85]: is this different that inspectRecords?

File: src/k2/dto/K23SIInspect.h (hunk 1 above)
Comment (jfunston) [reply src/k2/dto/K23SIInspect.h:45]: Yes. Are you worried about the header dependency?

File: src/k2/dto/K23SIInspect.h (hunk 2 above)
Comment (jfunston) [reply src/k2/dto/K23SIInspect.h:85]: The comment should say "all keys". The idea is to use it for the orphaned WI scan.

File: src/k2/dto/K23SIInspect.h (hunk 1 above)
Comment (iafuture) [reply src/k2/dto/K23SIInspect.h:45]: yes - are you moving DataRecord to DTO?

File: src/k2/dto/K23SIInspect.h (hunk 2 above)
Comment (iafuture) [reply src/k2/dto/K23SIInspect.h:85]: so i guess we should have InspectKey (all versions for 1 key) and InspectAllKeys(all versions for all keys) ?

File: src/k2/dto/K23SIInspect.h (hunk 1 above)
Comment (jfunston) [reply src/k2/dto/K23SIInspect.h:127]: ok, I moved it

File: src/k2/dto/K23SIInspect.h (hunk 2 above)
Comment (jfunston) [reply src/k2/dto/K23SIInspect.h]: I added InspectAllKeys, but it only returns the keys themselves. I figured that returning all the versions too would be too big for one request. Sound good?

**Comment by jfunston** [comment]
I added the RPC to get all TRHs. We can save the tracing support for a separate PR, I will create an issue for it.

File: src/k2/dto/K23SIInspect.h (hunk 2 above)
Comment (iafuture) [reply src/k2/dto/K23SIInspect.h]: makes sense

**Pull request merged**